cmd/manager/main.go

```diff
@@ -212,3 +224,23 @@ func setProfiler() {
 		}()
 	}
 }
+
+func lockHandler() (lockfile.Lockfile, error) {
+	lockFilePath, ok := os.LookupEnv("NMSTATE_INSTANCE_NODE_LOCK_FILE")
+	if !ok {
+		return "", errors.New("Failed to find NMSTATE_INSTANCE_NODE_LOCK_FILE ENV var")
+	}
+	log.Info("Try to take exclusive lock on file: %s", lockFilePath)
+	lockFileMutex, err := lockfile.New(lockFilePath)
+	if err != nil {
+		return lockFileMutex, errors.Wrapf(err, "failed to create lockFile for %s", lockFilePath)
+	}
+	for {
+		log.Info("try to get lock")
+		if err = lockFileMutex.TryLock(); err == nil {
+			log.Info("Successfully locked %s", lockFilePath)
+			return lockFileMutex, err
+		}
+		time.Sleep(10 * time.Second)
```

Comment:
wonder if there is already blocking Lock , and then retry isnt needed ?
(prints + context switch will be eliminated, also less code)

for example 
https://github.com/juju/fslock

func (*Lock) Lock
func (l *Lock) Lock() error
Lock locks the lock. This call will block until the lock is available.

Reply:
@oshoval  - thanks for the feedback.

https://github.com/juju/fslock is based on flock, AFAIK, there are some cases that flock doesn't work (NFS).  I think we should stick with solution that doesn't sue flock.

Reply:
thanks